internal/services/cognitive/cognitive_account_resource.go

```diff
@@ -50,277 +52,7 @@ func resourceCognitiveAccount() *pluginsdk.Resource {
-				Optional: true,
-				Default:  true,
-			},
-
-			"metrics_advisor_aad_client_id": {
-				Type:         pluginsdk.TypeString,
-				Optional:     true,
-				ForceNew:     true,
-				ValidateFunc: validation.IsUUID,
-			},
-
-			"metrics_advisor_aad_tenant_id": {
-				Type:         pluginsdk.TypeString,
-				Optional:     true,
-				ForceNew:     true,
-				ValidateFunc: validation.IsUUID,
-			},
-
-			"metrics_advisor_super_user_name": {
-				Type:         pluginsdk.TypeString,
-				Optional:     true,
-				ForceNew:     true,
-				ValidateFunc: validation.StringIsNotEmpty,
-			},
-
-			"metrics_advisor_website_name": {
-				Type:         pluginsdk.TypeString,
-				Optional:     true,
-				ForceNew:     true,
-				ValidateFunc: validation.StringIsNotEmpty,
-			},
-
-			"network_acls": {
-				Type:         pluginsdk.TypeList,
-				Optional:     true,
-				MaxItems:     1,
-				RequiredWith: []string{"custom_subdomain_name"},
-				Elem: &pluginsdk.Resource{
-					Schema: map[string]*pluginsdk.Schema{
-						"default_action": {
-							Type:     pluginsdk.TypeString,
-							Required: true,
-							ValidateFunc: validation.StringInSlice([]string{
-								string(cognitiveservicesaccounts.NetworkRuleActionAllow),
-								string(cognitiveservicesaccounts.NetworkRuleActionDeny),
-							}, false),
-						},
-						"ip_rules": {
-							Type:     pluginsdk.TypeSet,
-							Optional: true,
-							Elem: &pluginsdk.Schema{
-								Type: pluginsdk.TypeString,
-								ValidateFunc: validation.Any(
-									commonValidate.IPv4Address,
-									commonValidate.CIDR,
-								),
-							},
-							Set: set.HashIPv4AddressOrCIDR,
-						},
-						// TODO 3.0 - Remove below property
-						"virtual_network_subnet_ids": {
-							Type:          pluginsdk.TypeSet,
-							Optional:      true,
-							Computed:      true,
-							ConflictsWith: []string{"network_acls.0.virtual_network_rules"},
-							Deprecated:    "Deprecated in favour of `virtual_network_rules`",
-							Elem:          &pluginsdk.Schema{Type: pluginsdk.TypeString},
-						},
-
-						"virtual_network_rules": {
-							Type:          pluginsdk.TypeSet,
-							Optional:      true,
-							Computed:      true, // TODO -- remove this when deprecation resolves
-							ConflictsWith: []string{"network_acls.0.virtual_network_subnet_ids"},
-							ConfigMode:    pluginsdk.SchemaConfigModeAttr, // TODO -- remove in 3.0, because this property is optional and computed, it has to be declared as empty array to remove existed values
-							Elem: &pluginsdk.Resource{
-								Schema: map[string]*pluginsdk.Schema{
-									"subnet_id": {
-										Type:     pluginsdk.TypeString,
-										Required: true,
-									},
-
-									"ignore_missing_vnet_service_endpoint": {
-										Type:     pluginsdk.TypeBool,
-										Optional: true,
-										Default:  false,
-									},
-								},
-							},
-						},
-					},
-				},
-			},
-
-			"outbound_network_access_restrited": {
```

Comment:
Maybe I glossed over this too quickly and am missing something, but shouldn't this have a deprecation message instead of just a hard rename?

Reply:
Good call! I'll put the change behind the flag along with the rest.